src/quickfix.c

```diff
@@ -1010,6 +1014,21 @@ qf_parse_line(
 		fields->pattern[len + 4] = '$';
 		fields->pattern[len + 5] = NUL;
 	    }
+	    if ((i = (int)fmt_ptr->addr[10]) > 0)		/* %o */
+	    {
+		if (regmatch.startp[i] == NULL)
+		    continue;
+		len = (int)(regmatch.endp[i] - regmatch.startp[i]);
+		if (len >= fields->errmsglen)
```

Comment:
This is not correct. The length of the module string is CMDBUFFSIZE.
If "len" is greater than CMDBUFFSIZE, then len should be truncated.
You can copy the code from the block that handles "%s".